cmd/srcd/config/config.go

```diff
@@ -1,83 +1,71 @@
 package config
 
 import (
+	"io/ioutil"
 	"path/filepath"
 
+	"github.com/src-d/engine/api"
+
 	homedir "github.com/mitchellh/go-homedir"
 	"github.com/pkg/errors"
-	"github.com/sirupsen/logrus"
-	"github.com/spf13/viper"
-	"github.com/src-d/engine/api"
+	"gopkg.in/src-d/go-log.v1"
 	yaml "gopkg.in/yaml.v2"
 )
 
-// InitConfig reads the config file and ENV variables if set.
-func InitConfig(cfgFile string) {
+var configFile string
+
+// InitConfig reads the config file if set.
+func InitConfig(cfgFile string) error {
 	if cfgFile != "" {
 		// Use config file from the flag.
-		viper.SetConfigFile(cfgFile)
+		configFile = cfgFile
 	} else {
 		// Find home directory.
 		home, err := homedir.Dir()
 		if err != nil {
-			logrus.Fatalf("Could not detect home directory: %s", err)
+			return errors.Wrapf(err, "Could not detect home directory")
 		}
 
-		// Search config in $HOME/.srcd/ with name "config" (without extension).
-		viper.AddConfigPath(filepath.Join(home, ".srcd"))
-		viper.SetConfigName("config")
-	}
-
-	viper.AutomaticEnv() // read in environment variables that match
-
-	// If a config file is found, read it in.
-	err := viper.ReadInConfig()
-	if err != nil {
-		// viper returns ConfigFileNotFoundError only when the default config file
-		// is not found. For a wrong file path on --config it's an os.PathError
-		if _, ok := err.(viper.ConfigFileNotFoundError); !ok {
-			logrus.Fatalf("Error reading the config file %s: %s", viper.ConfigFileUsed(), err.Error())
-		}
+		configFile = filepath.Join(home, ".srcd", "config.yml")
 	}
 
-	checkConfig()
+	return checkConfig()
 }
 
-func checkConfig() {
-	actualConfigFile := viper.ConfigFileUsed()
-	if actualConfigFile == "" {
-		return
+func checkConfig() error {
```

Comment:
This is called only [here](https://github.com/src-d/engine/pull/425/files#diff-a727f8a7dbe085385785155fa0679272R32). I'd remove this, and move the logic inside `InitConfig`.

Reply:
done in f164c20